hail/src/test/scala/is/hail/expr/ir/MatrixIRSuite.scala

```diff
@@ -273,4 +273,35 @@ class MatrixIRSuite extends SparkSuite {
       assert(values.isSorted && values.length == 11, n -> shuffle)
     }
   }
+
+  @Test def testMatrixWrite() {
+    // Native Writer
+    val range = MatrixTable.range(hc, 11, 3, Some(10))
+    var path = tmpDir.createLocalTempFile(extension = "mt")
+    Interpret(MatrixWrite(range.ast, MatrixNativeWriter(path)))
+    val read = MatrixTable.read(hc, path)
+    assert(read.same(range))
+
+    // Export VCF
+    val vcf = hc.importVCF("src/test/resources/sample.vcf")
+    path = tmpDir.createLocalTempFile(extension = "vcf")
+    Interpret(MatrixWrite(vcf.ast, MatrixVCFWriter(path)))
+
+    // Export PLINK
+    val plinkPath = "src/test/resources/skip_invalid_loci"
+    val plink = hc.importPlink(plinkPath + ".bed", plinkPath + ".bim", plinkPath + ".fam", skipInvalidLoci = true)
+    val plinkIR = MatrixMapRows(plink.ast, InsertFields(Ref("va", plink.rvRowType), // FIXME: Make importPlink output varid as field name instead of rsid
+      FastIndexedSeq("varid" -> GetField(Ref("va", plink.rvRowType), "rsid"))))
+    path = tmpDir.createLocalTempFile()
+    Interpret(MatrixWrite(plinkIR, MatrixPLINKWriter(path)))
+
+    // Export GEN
+    val gen = hc.importGen("src/test/resources/example.gen", "src/test/resources/example.sample",
+      contigRecoding = Some(Map("01" -> "1")))
+    val genIR = MatrixMapCols(gen.ast, SelectFields(InsertFields(Ref("sa", gen.colType),
+      FastIndexedSeq("id1" -> Str(""), "id2" -> Str(""), "missing" -> F64(0.0))), FastIndexedSeq("id1", "id2", "missing")),
+      Some(FastIndexedSeq("id1"))) // FIXME: Make importGen add sample annotations from the sample file
```

Comment:
I think this is not a great place for this FIXME. Let's make it an issue and assign it to someone.